### PR TITLE
Add to docstring_test

### PR DESCRIPTION
**Pull request by twinters007**
Removes deprecated inspectargs

Adds checks for docstring and type hint

### CODE REVIEW

**Comment by twinters007** [comment]
Darn. Forgot that’s a thing

File: lib/pyfrc/tests/docstring_test.py [hunk 1]
```diff
@@ -74,12 +75,30 @@ def check_function(parent, fn, errors):
...
+            if match:
+                arg = match.group(2)
+                if arg not in args:
+                    print_fn_err("Param '%s' is documented but isn't a parameter for" % arg, parent, fn, errors)
+
+                if arg in annotations.keys() and type(annotations.get(arg)) is str:
+                    print_fn_err("Do not document %s in both the docstring and annotations" % arg, parent, fn, errors)
+
+                if arg in annotations.keys() and type(annotations.get(arg)) is type:
+                    print_fn_err("Do not document %s in both the docstring and annotations" % arg, parent, fn, errors)
```
Comment (auscompgeek) [inline lib/pyfrc/tests/docstring_test.py]: DRY

File: lib/pyfrc/tests/docstring_test.py [hunk 2]
```diff
@@ -1,140 +1,169 @@
...
+                if arg in annotations.keys() and type(annotations.get(arg)) is str:
+                    print_fn_err("Do not document %s in both the docstring and annotations" % arg, parent, fn, errors)
+
+                params.append(arg)
+
+            # :type arg: type
+            match = type_re.match(line)
+            if match:
+                arg = match.group(1)
+                if arg in annotations.keys() and type(annotations.get(arg)) is type:
```
Comment (auscompgeek) [inline lib/pyfrc/tests/docstring_test.py]: Never do `type(...) is ...`, that will almost certainly not do what you want. Use `isinstance`.

File: lib/pyfrc/tests/docstring_test.py [hunk 3]
```diff
@@ -1,140 +1,169 @@
...
+                    print_fn_err("Do not document %s in both the docstring and annotations" % arg, parent, fn, errors)
+
+                if arg in annotations.keys() and type(annotations.get(arg)) is type:
+                    print_fn_err("Do not document %s in both the docstring and annotations" % arg, parent, fn, errors)
+
+                params.append(arg)
+
+
+
+
```
Comment (auscompgeek) [inline lib/pyfrc/tests/docstring_test.py]: looks like someone smashed their return key

File: lib/pyfrc/tests/docstring_test.py (hunk 1 above)
Comment (twinters007) [reply lib/pyfrc/tests/docstring_test.py]: idk what dry means but i combined these two ifs

Comment (auscompgeek) [reply lib/pyfrc/tests/docstring_test.py]: (don't repeat yourself)

File: lib/pyfrc/tests/docstring_test.py (hunk 3 above)
Comment (twinters007) [reply lib/pyfrc/tests/docstring_test.py]: 👀

**Pull request merged**